### PR TITLE
SA1136 place each enum value on its own line

### PR DESCRIPTION
**Pull request by elachlan**
Relates to #7174
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1136.md

### CODE REVIEW

File: src/Tasks.UnitTests/ResolveComReference_Tests.cs [hunk 1]
```diff
@@ -753,7 +753,9 @@ public void AxReferenceKnowsItsRCWPiaExists_IncludeVersion()
             CheckAxReferenceRCWTlbExists(RcwStyle.PreexistingPia /* pass in the PIA reference */, true /* include version in the interop name */);
         }
 
-        private enum RcwStyle { GenerateTlb, PreexistingTlb, PreexistingPia };
+        private enum RcwStyle { GenerateTlb,
+            PreexistingTlb,
+            PreexistingPia };
```
Comment (sharwell) [inline src/Tasks.UnitTests/ResolveComReference_Tests.cs:758]: 📝 This one should be manually corrected

File: src/Tasks/GenerateManifestBase.cs [hunk 2]
```diff
@@ -16,8 +16,12 @@ namespace Microsoft.Build.Tasks
...
     public abstract class GenerateManifestBase : Task
     {
-        private enum AssemblyType { Unspecified, Managed, Native, Satellite };
-        private enum DependencyType { Install, Prerequisite };
+        private enum AssemblyType { Unspecified,
+            Managed,
+            Native,
+            Satellite };
+        private enum DependencyType { Install,
+            Prerequisite };
```
Comment (sharwell) [inline src/Tasks/GenerateManifestBase.cs:24]: 📝 This one should be manually corrected

File: src/Tasks/GenerateManifestBase.cs [hunk 3]
```diff
@@ -16,8 +16,12 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     public abstract class GenerateManifestBase : Task
     {
-        private enum AssemblyType { Unspecified, Managed, Native, Satellite };
-        private enum DependencyType { Install, Prerequisite };
+        private enum AssemblyType { Unspecified,
+            Managed,
+            Native,
+            Satellite };
```
Comment (sharwell) [inline src/Tasks/GenerateManifestBase.cs:22]: 📝 This one should be manually corrected

File: src/Tasks/GenerateApplicationManifest.cs [hunk 4]
```diff
@@ -17,7 +17,8 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     public sealed class GenerateApplicationManifest : GenerateManifestBase
     {
-        private enum _ManifestType { Native, ClickOnce }
+        private enum _ManifestType { Native,
+            ClickOnce }
```
Comment (sharwell) [inline src/Tasks/GenerateApplicationManifest.cs:21]: 📝 This one should be manually corrected

File: src/Tasks/FormatVersion.cs [hunk 5]
```diff
@@ -17,7 +17,8 @@ namespace Microsoft.Build.Tasks
     /// </comment>
     public sealed class FormatVersion : TaskExtension
     {
-        private enum _FormatType { Version, Path }
+        private enum _FormatType { Version,
+            Path }
```
Comment (sharwell) [inline src/Tasks/FormatVersion.cs:21]: 📝 This one should be manually corrected

File: src/Tasks.UnitTests/ResolveComReference_Tests.cs [hunk 6]
```diff
@@ -753,7 +753,11 @@ public void AxReferenceKnowsItsRCWPiaExists_IncludeVersion()
             CheckAxReferenceRCWTlbExists(RcwStyle.PreexistingPia /* pass in the PIA reference */, true /* include version in the interop name */);
         }
 
-        private enum RcwStyle { GenerateTlb, PreexistingTlb, PreexistingPia };
+        private enum RcwStyle {
+            GenerateTlb,
+            PreexistingTlb,
+            PreexistingPia
+        };
```
Comment (sharwell) [inline src/Tasks.UnitTests/ResolveComReference_Tests.cs:760]: 📝 All the enums in this pull request should use the consistent style for C# projects (Allman-style braces, no extraneous semicolon). I would encourage, but not require, the use of a trailing comma on the final enum item. Here's the result of all three suggestions applied to this case:

```suggestion
        private enum RcwStyle
        {
            GenerateTlb,
            PreexistingTlb,
            PreexistingPia,
        }
```

Comment (elachlan) [reply src/Tasks.UnitTests/ResolveComReference_Tests.cs]: > 

Done.

File: src/Build/Evaluation/Conditionals/Token.cs [hunk 7]
```diff
@@ -32,11 +32,24 @@ internal sealed class Token
...
+            Or,
+            EqualTo,
+            NotEqualTo,
+            Not,
+            Property,
+            String,
+            Numeric,
+            ItemList,
+            ItemMetadata,
+            Function,
```
Comment (Forgind) [inline src/Build/Evaluation/Conditionals/Token.cs:52]: extra semicolon; no trailing comma

File: src/Build/Graph/ProjectInterpretation.cs [hunk 8]
```diff
@@ -40,7 +40,9 @@ private ProjectInterpretation()
 
         internal enum ProjectType
         {
-            OuterBuild, InnerBuild, NonMultitargeting
+            OuterBuild,
+            InnerBuild,
+            NonMultitargeting
```
Comment (Forgind) [inline src/Build/Graph/ProjectInterpretation.cs:45]:
```suggestion
            NonMultitargeting,
```

File: src/Tasks.UnitTests/ResolveComReference_Tests.cs [hunk 9]
```diff
@@ -753,7 +753,11 @@ public void AxReferenceKnowsItsRCWPiaExists_IncludeVersion()
             CheckAxReferenceRCWTlbExists(RcwStyle.PreexistingPia /* pass in the PIA reference */, true /* include version in the interop name */);
         }
 
-        private enum RcwStyle { GenerateTlb, PreexistingTlb, PreexistingPia };
+        private enum RcwStyle {
```
Comment (Forgind) [inline src/Tasks.UnitTests/ResolveComReference_Tests.cs:756]:
```suggestion
        private enum RcwStyle
        {
```

File: src/Tasks/BootstrapperUtil/Product.cs [hunk 10]
```diff
@@ -12,7 +12,9 @@ namespace Microsoft.Build.Tasks.Deployment.Bootstrapper
 {
     internal enum CopyAllFilesType
     {
-        CopyAllFilesFalse, CopyAllFilesTrue, CopyAllFilesIfNotHomeSite
+        CopyAllFilesFalse,
+        CopyAllFilesTrue,
+        CopyAllFilesIfNotHomeSite
```
Comment (Forgind) [inline src/Tasks/BootstrapperUtil/Product.cs:17]:
```suggestion
        CopyAllFilesIfNotHomeSite,
```
and extra semicolon

File: src/Tasks/FormatVersion.cs [hunk 11]
```diff
@@ -17,7 +17,10 @@ namespace Microsoft.Build.Tasks
     /// </comment>
     public sealed class FormatVersion : TaskExtension
     {
-        private enum _FormatType { Version, Path }
+        private enum _FormatType {
```
Comment (Forgind) [inline src/Tasks/FormatVersion.cs:20]:
```suggestion
        private enum _FormatType
        {
```

File: src/Tasks/GenerateApplicationManifest.cs [hunk 12]
```diff
@@ -17,7 +17,10 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     public sealed class GenerateApplicationManifest : GenerateManifestBase
     {
-        private enum _ManifestType { Native, ClickOnce }
+        private enum _ManifestType {
```
Comment (Forgind) [inline src/Tasks/GenerateApplicationManifest.cs:20]:
```suggestion
        private enum _ManifestType
        {
```

File: src/Tasks/GenerateManifestBase.cs [hunk 13]
```diff
@@ -16,8 +16,17 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     public abstract class GenerateManifestBase : Task
     {
-        private enum AssemblyType { Unspecified, Managed, Native, Satellite };
-        private enum DependencyType { Install, Prerequisite };
+        private enum AssemblyType {
```
Comment (Forgind) [inline src/Tasks/GenerateManifestBase.cs:19]:
```suggestion
        private enum AssemblyType
        {
```

File: src/Tasks/GenerateManifestBase.cs [hunk 14]
```diff
@@ -16,8 +16,17 @@ namespace Microsoft.Build.Tasks
...
+            Unspecified,
+            Managed,
+            Native,
+            Satellite,
+        }
+
+        private enum DependencyType {
```
Comment (Forgind) [inline src/Tasks/GenerateManifestBase.cs:26]:
```suggestion
        private enum DependencyType
        {
```

File: src/Tasks/StrongNameUtils.cs [hunk 15]
```diff
@@ -18,7 +18,10 @@ namespace Microsoft.Build.Tasks
     /// </summary>
     internal enum StrongNameLevel
     {
-        None, DelaySigned, FullySigned, Unknown
+        None,
+        DelaySigned,
+        FullySigned,
+        Unknown
```
Comment (Forgind) [inline src/Tasks/StrongNameUtils.cs:24]:
```suggestion
        Unknown,
```
and extra semicolon

**Review by Forgind — APPROVED** [review]
Thanks!

File: src/Build/Evaluation/Conditionals/Token.cs [hunk 16]
```diff
@@ -32,13 +32,26 @@ internal sealed class Token
...
         internal enum TokenType
         {
-            Comma, LeftParenthesis, RightParenthesis,
-            LessThan, GreaterThan, LessThanOrEqualTo, GreaterThanOrEqualTo,
-            And, Or,
-            EqualTo, NotEqualTo, Not,
-            Property, String, Numeric, ItemList, ItemMetadata, Function,
-            EndOfInput
-        };
+            Comma,
```
Comment (Forgind) [inline src/Build/Evaluation/Conditionals/Token.cs:35]: Team triage: The original version was grouped logically; can you insert some spaces between those sets to make it easier to follow?

**Review by Forgind — APPROVED** [review]

**Review by ladipro — APPROVED** [review]

**Pull request merged**